src/settlement-chain/PayerReportManager.sol

```diff
@@ -504,4 +508,45 @@ contract PayerReportManager is IPayerReportManager, Initializable, Migratable, E
         // The signature is valid if the recovered signer is not `address(0)` and is the signer of the node.
         return !_isZero(signer_) && (signer_ == INodeRegistryLike(nodeRegistry).getSigner(nodeId_));
     }
+
+    function _enforceNodeIdsMatchRegistry(uint32[] calldata nodeIds_) internal view {
+        INodeRegistry.NodeWithId[] memory all = INodeRegistry(nodeRegistry).getAllNodes();
+        uint8 canonicalCount = INodeRegistry(nodeRegistry).canonicalNodesCount();
+
+        if (nodeIds_.length != canonicalCount) {
+            revert NodeIdsDoNotMatchRegistry(uint32(canonicalCount), uint32(nodeIds_.length));
+        }
+
+        uint256 j = 0; // index into nodeIds_ (submitted)
+        uint32 prev = 0; // for strictly-increasing check on submitted ids
+
+        for (uint256 i = 0; i < all.length; ) {
+            if (all[i].node.isCanonical) {
+                // Bounds guaranteed because nodeIds_.length == canonicalCount
+                uint32 expectedId = all[i].nodeId;
+                uint32 actualId = nodeIds_[j];
+
+                if (j > 0) {
+                    // strictly increasing check on submitted array
+                    if (actualId <= prev) revert UnorderedNodeIds();
+                }
+
+                if (actualId != expectedId) {
+                    revert NodeIdsDoNotMatchRegistry(expectedId, actualId);
```

Comment:
The interface for this suggests that the two values should be counts, but here you are using IDs

Reply:
fixed